backend/src/users/controllers.ts

```diff
@@ -563,6 +562,58 @@ const editHours = async (req: Request, res: Response) => {
   }
 };
 
+/**
+ * Returns sorted Users based on one key
+ * @returns promise with user or error
+ */
+const getUsersSorted = async (req: Request, res: Response) => {
+  // #swagger.tags = ['Users']
+  const query = req.query.sort as string;
+  const querySplit = query.split(":");
+  const key: string = querySplit[0];
+  const order = querySplit[1];
+
+  try {
+    if (key == "email") {
+      const users = await prisma.user.findMany({
+        orderBy: [{ email: order == "asc" ? "asc" : "desc" }],
+      });
+      res.status(200).json(users);
+    } else if (key == "hours") {
+      const users = await prisma.user.findMany({
+        orderBy: [{ hours: order == "asc" ? "asc" : "desc" }],
+      });
+      res.status(200).json(users);
+    } else if (key == "firstName") {
+      const users = await prisma.user.findMany({
+        orderBy: {
+          profile: {
+            firstName: order == "asc" ? "asc" : "desc",
+          },
+        },
+        include: {
+          profile: true,
+        },
+      });
```

Comment:
Observe that this code is precisely the same as the code on lines 600-616, you could easily pull this out into a function that simply takes in `order` as a parameter, as such:
```javascript
const findUsers = async (order : string) => {
    return prisma.user.findMany({
        orderBy: {
            profile: {
                firstName: order
            },
        },
        [...other code here]
    })
}
```

Or, with some syntactic sugar:
```javascript
const findUsers = async (order : string) =>
   prisma.user.findMany({
        orderBy: {
            profile: {
                firstName: order
            },
        },
        [...other code here]
    })
```

One thing that could be confusing is the fact that we don't `await` the async call in the function. This is because the function itself is async, so we simply `await` the overall function call via `const users = await findUsers();`

This is _precisely_ what controllers are meant to be used for (as in, functions like `findUsers()`), to ensure code re-use. This controllers file is not structured properly -- I've already mentioned this to Akin and Jason, but it's vital that you remind them to resolve this issue with a refactor.